lib/stat_tracker.rb

```diff
@@ -50,12 +50,243 @@ def percentage_ties
     (ties / game_collection.games.length.to_f).round(2)
   end
 
-  def count_of_games_by_season
+  def worst_fans
+    home_w = hoa_wins_by_team("home")
+    away_w = hoa_wins_by_team("away")
+    teams = away_w.select do |team, away_wins|
+      home_w[team] < away_wins
+    end.keys
+    teams.map { |team_id| team_name_by_id(team_id) }
+  end
+
+  def lowest_scoring_home_team
+    home_games = hoa_games_by_team("home")
+    home_goals = hoa_goals_by_team("home")
+    home_goals_per_game = {}
+    home_goals.each do |team_id, total_home_goals|
+      next if total_home_goals == 0
+      home_goals_per_game[team_id] = total_home_goals / hoa_games_by_team("home")[team_id].to_f
+    end
+    team_name_by_id(home_goals_per_game.key(home_goals_per_game.values.min))
+  end
+
+  def highest_scoring_home_team
+    home_games = hoa_games_by_team("home")
+    home_goals = hoa_goals_by_team("home")
+    home_goals_per_game = {}
+    home_goals.each do |team_id, total_home_goals|
+      next if total_home_goals == 0
+      home_goals_per_game[team_id] = total_home_goals / hoa_games_by_team("home")[team_id].to_f
+    end
+    team_name_by_id(home_goals_per_game.key(home_goals_per_game.values.max))
+  end
+
+  def best_defense
+    allowed = all_goals_allowed_by_team
+    games = total_games_by_team
+    average_goals_allowed = {}
+    allowed.each do |team_id, goals_allowed|
+      average_goals_allowed[team_id] = goals_allowed / total_games_by_team[team_id].to_f
+    end
+    team_name_by_id(average_goals_allowed.key(average_goals_allowed.values.min))
+  end
+
+  def worst_coach(season)
+    # averages = {}
+    # wins_in_season(season).each do |team_id, wins|
+    #   averages[team_id] = wins / games_by_team_by_season(season)[team_id].to_f
+    # end
+    # averages
+    # require "pry"; binding.pry
+    # head_coaches(season)[averages.key(averages.values.min)]
+  end
+
+  ###### move these methods somewhere else
```

Comment:
Great work on the helper methods. Utilizing inheritance will help clean up the stat tracker, and abide by SRP.